private/bufpkg/bufcheck/buflint/internal/buflintvalidate/well_known_type.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package buflintvalidate
+
+import "strings"
+
+// WellKnownTypePackage is the proto package name where all Well Known Types
+// currently reside.
+const WellKnownTypePackage string = "google.protobuf."
+
+// WellKnownType (WKT) encapsulates the Name of a Message from the
+// `google.protobuf` package. Most official protoc plugins special case code
+// generation on these messages.
+type WellKnownType string
+
+// 1-to-1 mapping of the WKT names to WellKnownTypes.
+const (
+	// UnknownWKT indicates that the type is not a known WKT. This value may be
+	// returned erroneously mapping a Name to a WellKnownType or if a WKT is
+	// added to the `google.protobuf` package but this library is outdated.
+	UnknownWKT WellKnownType = "Unknown"
+
+	AnyWKT         WellKnownType = "Any"
+	DurationWKT    WellKnownType = "Duration"
+	EmptyWKT       WellKnownType = "Empty"
+	StructWKT      WellKnownType = "Struct"
+	TimestampWKT   WellKnownType = "Timestamp"
+	ValueWKT       WellKnownType = "Value"
+	ListValueWKT   WellKnownType = "ListValue"
+	DoubleValueWKT WellKnownType = "DoubleValue"
+	FloatValueWKT  WellKnownType = "FloatValue"
+	Int64ValueWKT  WellKnownType = "Int64Value"
+	UInt64ValueWKT WellKnownType = "UInt64Value"
+	Int32ValueWKT  WellKnownType = "Int32Value"
+	UInt32ValueWKT WellKnownType = "UInt32Value"
+	BoolValueWKT   WellKnownType = "BoolValue"
+	StringValueWKT WellKnownType = "StringValue"
+	BytesValueWKT  WellKnownType = "BytesValue"
+)
+
+var lookupTable = map[string]WellKnownType{
```

Comment:
`stringToWellKnownType`, similar to rest of codebase

Reply:
resolved in [182d2e2](https://github.com/bufbuild/buf/pull/2377/commits/182d2e21541d6084e75fd345150fa990cde60d62)